sycl/include/sycl/types.hpp

```diff
@@ -2085,298 +2085,119 @@ __SYCL_RELLOGOP(||)
 } // __SYCL_INLINE_VER_NAMESPACE(_V1)
 } // namespace sycl
 
-#ifdef __SYCL_USE_EXT_VECTOR_TYPE__
```

Comment:
Note for reviewers: previous approach heavily relied on macro and aliases and looked like this:
For each type and vec size we defined an internal alias `__{type}{size}_t` (or something like that) using macro magic. Each such alias would point to an actual type to use as a `vec` storage type. Then we had a bunch of struct template specializations which would reference a particular "proxy" alias (specializations were generated using macro again).

The new approach defines `VecStorageImpl` struct template, which provides actual data type to use as a storage type. On top of it we have a few `VecStorage` struct template specializations, which select the proper `VecStorageImpl` to use.

The main purpose of this patch is to stop referencing `sycl::cl_*` types which do not exists anymore according to the SYCL spec.

The main difference between implementations is that new implementation directly uses `::cl_*` types from `CL/OpenCL.h` header instead of relying on `sycl::cl_*` type aliases when `ext_vector_type` is not available. Also, new approach uses less macro-magic and relies more on struct template specializations.